### ISSUE TITLE
sometimes gets stuck on one channel when its supposed to scan

### ISSUE THREAD

**Issue by steeley** [br101/horst#95]
Another problem....

when started in scan mode (with er without interface) sometimes seems to get stuck on
one channel and it won't scan. have to reboot to fix. then of course it will come back again
at some point....

Platform: Raspberry Pi 3 B
OS:
Raspbian GNU/Linux 9 (stretch)
Linux pi3 4.14.79-v7+ #1159 SMP armv7l GNU/Linux

**Comment by br101** [comment]
what version of horst? can you compile with debug output (make DEBUG=1)?

**Comment by steeley** [comment]
I'm using Horst 5.1
horst -s -u 12

it seems to be ok 90% of the time. I've spent hours messing with it and the failure
is very random. it seems to be if you turn off scan and set a channel sometimes
it gets stuck on that channel. Sometimes it won't switch to that channel.
Sometimes set channel to zero and turn on scan and it is ok. sometimes not.

It may be an issue with the wifi stick(??) - TP-Link TL-W727N (Ralink MT7601U)
Once it gets stuck only a reboot clears it.
Then I swapped to a Ralink RT3572 stick and it was still stuck until I rebooted
. Horst did show' 211 message failed:object busy' 
If I reboot it the problem is fixed for a while. I think maybe the RT3572 is better - I will
use that and see if it fails.
 tried with DEBUG=1 -  didn't see any difference.
